//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

// $title: SGX Local Assertion Authority Config
// $overview: Protobuf definition for SGX Local Assertion Authority Config.
// $location: https://asylo.dev/docs/reference/proto/identity/asylo.sgx_local_assertion_authority_config.v1.html
// $front_matter: redirect_from:
// /docs/reference/proto/identity/sgx_local_assertion_authority_config.html

// This package defines the configuration for an SGX local assertion authority,
// which is included in an EnclaveAssertionAuthorityConfig message.
package asylo;

// Configuration for an SGX local assertion authority.
message SgxLocalAssertionAuthorityConfig {
  // A 16-byte value that defines the attestation domain to which this SGX local
  // assertion authority belongs. This value is used to determine whether two
  // enclaves are running on the same hardware. SGX enclaves running on the same
  // hardware can use SGX local attestation.
  //
  // An attestation domain is expected to be unique to a particular SGX machine
  // (i.e., two different SGX machines should have different attestation
  // domains). It can be thought of as the machine's UUID.
  optional bytes attestation_domain = 1;
}
